login-app/src/index.js

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+import App from './components/App';
+
+import { BrowserRouter as Router } from 'react-router-dom'
+import { Provider } from 'react-redux'
+import { createStore, applyMiddleware  } from 'redux'
+
+import rootReducer from './reducers'
+import thunk from 'redux-thunk'
+import logger from 'redux-logger'
+
+const store = createStore(
+    rootReducer,
+    applyMiddleware(thunk, logger)
+)
```

Comment:
While you named it root reducer, it is not a true combined reducer.  For it to be a combined reducer, you need to have 2 or more separate reducers and combine them using the { combineReducer } from 'redux'.

Then when you are using the combineReducer, your state will be an object of objects.  Each inner object represents a separate reducer's state.  Right now you only have 1 reducer that is not combined and thus your state is just an object.

https://redux.js.org/api/combinereducers is the docs for more examples.

Reply:
Your code has
```
state = {
    friends: [],
    error: '',
    errorStatusCode: null,
    fetchingData: false,
    isLoggingIn: false
}
```

Reply:
a combined reducer state looks like
```
combinedState = {
   login : {username: ""},
   friends: {
    friends: [],
    error: '',
    errorStatusCode: null,
    fetchingData: false,
    isLoggingIn: false
   }
}
```